activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -90,16 +90,20 @@ def expand_from_hash(attributes)
             queries.reduce(&:or)
           elsif table.aggregated_with?(key)
             mapping = table.reflect_on_aggregation(key).mapping
-            queries = Array.wrap(value).map do |object|
-              mapping.map do |field_attr, aggregate_attr|
-                if mapping.size == 1 && !object.respond_to?(aggregate_attr)
-                  build(table.arel_attribute(field_attr), object)
-                else
+            if mapping.length == 1
+              column_name, aggr_attr = mapping.first
+              values = Array.wrap(value).map do |object|
+                object.respond_to?(aggr_attr) ? object.send(aggr_attr) : object
```

Comment:
Use `public_send` instead since `respond_to?` doesn't include private/protected methods by default.

https://ruby-doc.org/core-2.6/Object.html#method-i-respond_to-3F

```suggestion
                object.respond_to?(aggr_attr) ? object.public_send(aggr_attr) : object
```